### PR TITLE
add androidx support

### PR DESCRIPTION
**Pull request by AndreasBoehm**
This PR adds androidx support and closes https://github.com/sockeqwe/fragmentargs/issues/102

To make it easier to work with support fragments, androidx fragments and framework fragments in a single project but still using `NonNull` and `Nullable` annotations we are trying to find the available Annotation Classes during processor setup. By doing so we don't need to worry about the exact fragment type while processing.

### CODE REVIEW

File: processor/src/main/java/com/hannesdorfmann/fragmentargs/processor/ArgProcessor.java [hunk 1]
```diff
@@ -729,12 +751,26 @@ private void scanForAnnotatedFragmentClasses(RoundEnvironment env,
...
+            }
+        }
 
-        return (fragmentType != null && typeUtils.isSubtype(classElement.asType(),
-                fragmentType.asType())) || (supportFragmentType != null && typeUtils.isSubtype(
-                classElement.asType(), supportFragmentType.asType()));
+        return false;
+    }
+
+    private boolean isClassAvailable(String className) {
```
Comment (sockeqwe) [inline processor/src/main/java/com/hannesdorfmann/fragmentargs/processor/ArgProcessor.java]: Not sure if this works always. Should we check if `TypeElement` exists or is that done someone before calling this and i have overlooked it?

Comment (AndreasBoehm) [reply processor/src/main/java/com/hannesdorfmann/fragmentargs/processor/ArgProcessor.java]: Works fine for me but i am still not sure whats the best way to check if a Class exists. I haven't tried using `TypeElement` for it yet. Will investigate and update my PR accordingly.

Comment (sockeqwe) [reply processor/src/main/java/com/hannesdorfmann/fragmentargs/processor/ArgProcessor.java]: I will take a look at this at during the weekend

**Comment by sockeqwe** [comment]
Thanks. I will do some final testing and release it. Thank you so much! I owe you at least one beer 🍺

**Comment by MarcinOrlowski** [comment]
Bit off-topic, but is this release going to be 4.0.0 finally? I am still using your 4.0.0-SNAPSHOT from Sonatype due to features you added there, so please do the merge if that was not the former intention.

**Comment by sockeqwe** [comment]
4.0.0 final

Marcin Orlowski <notifications@github.com> schrieb am Do., 31. Mai 2018,
07:34:

**Comment by carvaq** [comment]
Hi

We are migrating our app to AndroidX, with your current released version we get an error. When do you think you'll be able to release these changes? 

Cheers

**Comment by sockeqwe** [comment]
I'm about to release 4.0 right now. It's just that I have a new pc an strugle a bit setting up all maven credentials for authentication and signing  but should be published in the next couple of hours

**Comment by carvaq** [comment]
Awesome, thank you very much!

**Comment by sockeqwe** [comment]
It's published as `4.0.0-RC1`. It just takes some more time (1-2 hours) to sync to maven central.

So should be available in a few hours to everyone. Please let me know if you face any issue. If that is not the case, I will publish a final `4.0.0` version end of next week.

**Pull request merged**